src/solvers/smt2_incremental/smt2_incremental_decision_procedure.cpp

```diff
@@ -62,18 +62,16 @@ get_problem_messages(const smt_responset &response)
 ///   `convert_expr_to_smt`. This is because any sub expressions which
 ///   `convert_expr_to_smt` translates into function applications, must also be
 ///   returned by this`gather_dependent_expressions` function.
-static std::vector<exprt> gather_dependent_expressions(const exprt &expr)
-{
-  std::vector<exprt> dependent_expressions;
-  expr.visit_pre([&](const exprt &expr_node) {
-    if(
-      can_cast_expr<symbol_exprt>(expr_node) ||
-      can_cast_expr<array_exprt>(expr_node))
-    {
-      dependent_expressions.push_back(expr_node);
-    }
-  });
-  return dependent_expressions;
+/// \note `symbol_exprt`, `array_exprt` and `nondet_symbol_exprt` add dependant expressions.
+static std::vector<exprt> gather_dependent_expressions(const exprt &expr) {
+    std::vector<exprt> dependent_expressions;
+    expr.visit_pre([&](const exprt &expr_node) {
+        if (can_cast_expr<symbol_exprt>(expr_node) || can_cast_expr<array_exprt>(expr_node) ||
+            can_cast_expr<nondet_symbol_exprt>(expr_node)) {
```

Comment:
This is really hard to see the actual diff here? Is it only the above line? `can_cast_expr<nondet_symbol_exprt>(expr_node)`?

Reply:
No it's clearer (formatting...)